packages/autocomplete-plugin-recent-searches/src/createRecentSearchStore.ts

```diff
@@ -0,0 +1,64 @@
+function isLocalStorageSupported() {
+  const key = '__TEST_KEY__';
+
+  try {
+    localStorage.setItem(key, '');
+    localStorage.removeItem(key);
+
+    return true;
+  } catch (error) {
+    return false;
+  }
+}
+
+function createStorage(key) {
+  if (isLocalStorageSupported() === false) {
+    return {
+      setItem() {},
+      getItem() {
+        return [];
+      },
+    };
+  }
+
+  return {
+    setItem(item) {
+      return window.localStorage.setItem(key, JSON.stringify(item));
+    },
+    getItem() {
+      const item = window.localStorage.getItem(key);
+
+      return item ? JSON.parse(item) : [];
+    },
+  };
+}
+
+export function createRecentSearchStore({ key, limit = 5 }) {
```

Comment:
We don't need to default `limit` at this level since it is at a higher level. Let's make our APIs as explicit as possible.

Reply:
done https://github.com/algolia/autocomplete.js/pull/316/commits/90fc79061c0beeb95cceedd446d8f1e2445a8b21